### ISSUE TITLE
Security: No login required

### ISSUE THREAD

**Issue by danpalmer** [monzo/response#127]
Currently response does not enforce login in the UI views or the API resources. This goes somewhat against the Django convention of being secure and safe by default.

It's possible that Monzo are deploying this internally behind a SSO proxy service, but for many users this might be prohibitively costly to set up.

I'm not sure what the right form of authentication is here. My recommendation is that response implements a `response_login_required` decorator, that extends the Django provided `login_required` decorator but can optionally disable it based on a configuration option. This option would be on by default, but disabling it provides a quick upgrade path for those installations that can be entirely unauthenticated.

This decorator would be applied to all URLs managed by response and Django Rest Framework, except for those used in the Slack integration.

**Comment by milesbxf** [comment]
Internally, we're actually using Django auth using python-social-auth with an SSO provider - we'd definitely want to give users the flexibility to set up Django auth however they want.

In the `demo/settings/base.py` we also configure the Django Rest Framework default permission class:
```
REST_FRAMEWORK = {
    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination',
    'PAGE_SIZE': 100,
    # Use Django's standard `django.contrib.auth` permissions,
    # or allow read-only access for unauthenticated users.
    'DEFAULT_PERMISSION_CLASSES': [
        'rest_framework.permissions.IsAuthenticated'
    ]
}
```

But I totally get the point that both of these things actively require users to set them up, and we should try and have secure defaults. I really like the idea of the overridable decorator - feel free to raise a PR if you've got a good idea of how this could work 👍

**Comment by danpalmer** [comment]
@milesbxf thanks for the details, we're actually using exactly the same library, at least for the non-response views at the moment.

This is my limited knowledge of DRF showing, I wasn't aware that you could configure auth at the DRF layer like this, we'll definitely do that.

I think auth is still necessary for the `/incident/<id>` view. I can have a stab at the login decorator for that and future views.

**Comment by danpalmer** [comment]
@milesbxf I've put this up in PR #129.

**Comment by milesbxf** [comment]
@danpalmer now that's merged, are you happy to close this?

**Comment by danpalmer** [comment]
Addressed by #129. I think there's more we could do to perhaps encapsulate some of the DRF configuration, but that's a wider discussion for another time.
